pageserver/src/tenant/tasks.rs

```diff
@@ -244,14 +244,12 @@ pub(crate) async fn random_init_delay(
 ) -> Result<(), Cancelled> {
     use rand::Rng;
 
+    if period == Duration::ZERO {
+        return Ok(());
+    }
+
```

Comment:
Maybe extract this to a separate PR. It seems worth drawing some extra attention to.

Reply:
https://github.com/neondatabase/neon/pull/3914